contractcourt/channel_arbitrator.go

```diff
@@ -461,6 +460,29 @@ func (c *ChannelArbitrator) relaunchResolvers(confirmedHTLCs []channeldb.HTLC) e
 	}
 	commitHash := contractResolutions.CommitHash
 
+	// In prior versions of lnd, the information needed to supplement the
+	// resolvers was found in the chain action map, which is now
+	// deprecated.  As a result, if the commitSet is nil (an older node
+	// with unresolved HTLCs at time of upgrade), then we'll use the chain
+	// action information in place. The chain actions may exclude some
+	// information, but we cannot recover it for these older nodes at the
+	// moment.
+	var confirmedHTLCs []channeldb.HTLC
+	if commitSet != nil {
+		confirmedHTLCs = commitSet.HtlcSets[*commitSet.ConfCommitKey]
+	} else {
+		chainActions, err := c.log.FetchChainActions()
```

Comment:
What is the long term plan to get rid of chain actions? Can they be migrated to confirmed commit sets? 

One nice (or maybe not so nice) possibilty is to do an in-flight migration in the channel arbitrator, if it is difficult to write a migration without context. Because we now only do major-to-major upgrades, we can remove the in-flight migration code again for the next major release. Only we need to be sure that it actually ran successfully. Not sure how to do that.

Reply:
Chain actions are deprecated, only those that upgraded with unresolved contracts need to deal with them. We can't directly port them over, as with the on-disk state, it isn't clear _which_ commitment the chain actions were created from (as the trigger that initiated the state transition isn't persisted anywhere). At this stage, the resolvers have already been created as well. 

AFAIK, the success resolver is the only one that can be potentially cancelled back by the invoice registry due to having a zero value amount for example.

Reply:
Ok, so best effort here. It would be nice to at some point be able to remove the chain actions bucket completely though.

Reply:
Agreed, we can look at it for 0.9 perhaps? Or even make our update policy even more aggressive: there's a min version, but also you can't update if you have pending contract resolutions. Ofc, this makes a pretty big trade off as far as agility if we need to roll out critical fixes in the future.